test/extended/router/externalCertificate.go

```diff
@@ -0,0 +1,682 @@
+				g.Context("to use new external certificate, but secret does not exist", func() {
+					// do not create new secret
+					g.It("route update is rejcted", func() {
+						g.By("Updating the route to use new external certificate")
+						routePatch := fmt.Sprintf(`{"spec":{"tls":{"externalCertificate":{"name":"%s"}}}}`, newSecretName)
+						_, err = oc.RouteClient().RouteV1().Routes(oc.Namespace()).Patch(
+							context.Background(), routeToTest.Name, types.MergePatchType, []byte(routePatch), metav1.PatchOptions{},
+						)
+						o.Expect(err).To(o.HaveOccurred())
+						o.Expect(err.Error()).To(o.ContainSubstring(fmt.Sprintf(`Not found: "secrets \"%s\" not found"`, newSecretName)))
+					})
+				})
+				g.Context("to use same external certificate", func() {
+					g.BeforeEach(func() {
+						g.By("Adding some label to trigger route update")
+						routePatch := `{"metadata":{"labels":{"app":"myapp","key":"value"}}}`
+						_, err = oc.RouteClient().RouteV1().Routes(oc.Namespace()).Patch(
+							context.Background(), routeToTest.Name, types.MergePatchType, []byte(routePatch), metav1.PatchOptions{},
+						)
+						o.Expect(err).NotTo(o.HaveOccurred())
+					})
+
+					g.It("then also the route is reachable", func() {
+						g.By("Sending https request")
+						hostName, err := getHostnameForRoute(oc, routeToTest.Name)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						resp, err := httpsGetCall(hostName, rootDerBytes)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						o.Expect(resp).Should(o.ContainSubstring("Hello OpenShift"))
+					})
+
+				})
+				g.Context("to use same external certificate, but RBAC permissions are dropped", func() {
+					g.BeforeEach(func() {
+						g.By("Deleting RBAC permissions")
+						err = oc.KubeClient().RbacV1().RoleBindings(oc.Namespace()).Delete(context.Background(), SECRET_READER_ROLE_BINDING, metav1.DeleteOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+					})
+
+					g.It("route update is rejcted", func() {
+						g.By("Adding some label to trigger route update")
+						routePatch := `{"metadata":{"labels":{"app":"myapp","key":"value"}}}`
+						_, err = oc.RouteClient().RouteV1().Routes(oc.Namespace()).Patch(
+							context.Background(), routeToTest.Name, types.MergePatchType, []byte(routePatch), metav1.PatchOptions{},
+						)
+						o.Expect(err).To(o.HaveOccurred())
+						o.Expect(err.Error()).To(o.And(
+							o.ContainSubstring("Forbidden: router serviceaccount does not have permission to get this secret"),
+							o.ContainSubstring("Forbidden: router serviceaccount does not have permission to watch this secret"),
+							o.ContainSubstring("Forbidden: router serviceaccount does not have permission to list this secret")),
+						)
+					})
+				})
+				g.Context("to remove and again re-add the same external certificate", func() {
+					g.BeforeEach(func() {
+						g.By("Updating the route to remove the external certificate reference")
+						routePatch := `{"spec": {"tls": null}}`
```

Comment:
not sure, how easy it is to test out:
unsetting this field `spec.tls`, after this we should still do a HTTPS call with the same route
use InsecureVerifyTLS and verify that the certificate is NOT the same (i.e. a different CA / something on those lines).